### PR TITLE
YLP-1916 Migrate tooltips from viz to dumb

### PR DESCRIPTION
**Pull request by celia-rouis**
Migrated almost all daily tooltips from `viz` package to `dumb` package

Tooltips migrated:
- CbgTooltip (moved to BloodGlucoseTooltip)
- ConfidentialTooltip
- FoodTooltip
- ParameterTooltip
- PhysicalTooltip
- SmbgTooltip (moved to BloodGlucoseTooltip)

### CODE REVIEW

File: packages/medical-domain/src/domains/repositories/medical/MedicalDataService.ts [hunk 1]
```diff
@@ -138,6 +181,11 @@ class MedicalDataService {
   }
 
   private join(): void {
+    this.joinBolus()
```
Comment (TimCanu) [inline packages/medical-domain/src/domains/repositories/medical/MedicalDataService.ts:184]: I don't know any of this code, can you explain it to me?

Comment (celia-rouis) [reply packages/medical-domain/src/domains/repositories/medical/MedicalDataService.ts:184]: Yes 😄 This intends to fix a bug in the `ReservoirChange` object, where the `ReservoirChange.pump` property was not filled.

When the medical data is received, it is processed through the `add()` method (see line 418). Each object type of the medical data is formatted in the `normalize()` method, and then, if needed, some jointures between 2 objects can be done in the `join()` method. It was already the case for the `bolus` data, so I added another method to enrich each `ReservoirChange` object with the data from the `PumpSettings`.

In order to fill the `ReservoirChange.pump` object, we want to take the most recent `PumpSettings`, and retrieve only the `PumpSettings.pumpConfig` property. 

That's my understanding of the subject but I may have misunderstood some things 👼

File: packages/dumb/src/components/tooltips/confidential-tooltip/confidential-tooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  side: Side
+}
+
+const t = i18next.t.bind(i18next)
+
+export const ConfidentialTooltip: FunctionComponent<ConfidentialTooltipProps> = (props) => {
+  const { position, side } = props
+
+  const tooltipParams = {
+    position,
```
[the remark this answers is not in the document]
Comment (celia-rouis) [reply packages/dumb/src/components/tooltips/confidential-tooltip/confidential-tooltip.tsx]: I did at first, but now that I've done some refactoring it does not seem relevant anymore. I removed it on all files

**Review by TimCanu — APPROVED** [review]

File: packages/medical-domain/src/domains/repositories/medical/MedicalDataService.ts [hunk 3]
```diff
@@ -157,6 +205,20 @@ class MedicalDataService {
     })
   }
 
+  private joinReservoirChanges(): void {
+    const sortedDescPumpSettings = this.medicalData.pumpSettings.sort((pumpSettings1: PumpSettings, pumpSettings2: PumpSettings) => {
+      const date1 = Number(new Date(pumpSettings1.normalTime))
+      const date2 = Number(new Date(pumpSettings2.normalTime))
+      return date2 - date1
```
Comment (CedricPerotto) [inline packages/medical-domain/src/domains/repositories/medical/MedicalDataService.ts:212]: you could use directly the pumpSettings1.epoch fields (timestamp value)
```suggestion
      return pumpSettings2.epoch- pumpSettings1.epoch
```

Or use the sortDatum private method of this class (sort order is different but it does the same thing).

File: packages/dumb/src/models/blood-glucose.model.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+export interface BgClassesRaw {
+  'very-low': { boundary: number }
+  low: { boundary: number }
+  target: { boundary: number }
+  high: { boundary: number }
+  'very-high': { boundary: number }
+}
+
```
[the remark this answers is not in the document]
Comment (celia-rouis) [reply packages/dumb/src/models/blood-glucose.model.ts]: I exported it 👍

File: packages/medical-domain/src/domains/repositories/medical/MedicalDataService.ts (hunk 3 above)
Comment (celia-rouis) [reply packages/medical-domain/src/domains/repositories/medical/MedicalDataService.ts]: I hadn't seen the `sortDatum()` method, it's better now that I reuse it

**Pull request merged**